components/feature/autofill/src/main/java/mozilla/components/feature/autofill/authenticator/Authenticator.kt

```diff
@@ -0,0 +1,61 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.autofill.authenticator
+
+import android.content.Context
+import androidx.fragment.app.FragmentActivity
+import mozilla.components.feature.autofill.AutofillConfiguration
+
+/**
+ * Shared interface to support multiple authentication methods.
+ */
+internal interface Authenticator {
+    /**
+     * Shows an authentication prompt and will invoke [callback] once authentication succeeded or
+     * failed.
+     */
+    fun prompt(activity: FragmentActivity, callback: Callback)
+
+    /**
+     * For passing an activity launch result to the authenticator.
+     */
+    fun onActivityResult(requestCode: Int, resultCode: Int)
```

Comment:
Would it make sense to lift "canEnroll" and "isAvailable" (and maybe add "requestEnroll") into here, as well? E.g. to allow application to just deal with an "Authenticator" instance that we pick for it based on the best that's available. Or do you think the UX/etc around actual enrolment would have to live within the apps..?

Reply:
Yeah, maybe, https://github.com/mozilla-mobile/android-components/issues/9756 is a good time to look at this.